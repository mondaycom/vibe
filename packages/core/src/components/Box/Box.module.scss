// Monday ui styles dependencies
@import "~monday-ui-style/dist/mixins";
@import "./utilities";

.box {
  overflow: hidden;

  &.scrollable {
    overflow: auto;
    @include scroller;
  }
}

// Generating Utility class locally for Box component
@include utility-class-factory($utilities-modules, $export: "modules");
