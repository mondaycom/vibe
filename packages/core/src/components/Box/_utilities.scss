@use "sass:map";
@use "sass:string";
@use "sass:list";
@import "~monday-ui-style/dist/functions";

/*  stylelint-disable scss/at-if-no-null */
$utility-spacing-vars: (
  0: 0,
  xs: var(--spacing-xs),
  small: var(--spacing-small),
  medium: var(--spacing-medium),
  large: var(--spacing-large),
  xl: var(--spacing-xl),
  xxl: var(--spacing-xxl),
  xxxl: var(--spacing-xxxl)
);

$utility-border-colors-vars: (
  uiBorderColor: var(--ui-border-color),
  layoutBorderColor: var(--layout-border-color)
);

$utility-colors-vars: (
  primaryTextColor: var(--primary-text-color),
  textColorOnInverted: var(--text-color-on-inverted),
  secondaryTextColor: var(--secondary-text-color)
);

$utility-background-colors-vars: (
  primaryBackgroundColor: var(--primary-background-color),
  secondaryBackgroundColor: var(--secondary-background-color),
  greyBackgroundColor: var(--grey-background-color),
  allgreyBackgroundColor: var(--allgrey-background-color),
  invertedColorBackground: var(--inverted-color-background)
);

$utilities-modules: (
  "opacity": (
    property: opacity,
    values: (
      disabled: var(--disabled-component-opacity)
    )
  ),
  "border": (
    property: border,
    values: (
      null: var(--border-width) var(--border-style) var(--ui-border-color)
    )
  ),
  "borderColor": (
    property: border-color,
    class: borderColor,
    values: $utility-border-colors-vars
  ),
  "rounded": (
    property: border-radius,
    class: rounded,
    values: (
      0: 0,
      small: var(--border-radius-small),
      medium: var(--border-radius-medium),
      big: var(--border-radius-big)
    )
  ),
  "shadow": (
    property: box-shadow,
    class: shadow,
    values: (
      xs: var(--box-shadow-xs),
      small: var(--box-shadow-small),
      medium: var(--box-shadow-medium),
      large: var(--box-shadow-large)
    )
  ),
  "margin": (
    property: margin,
    class: m,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "marginX": (
    property: margin-right margin-left,
    class: mx,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "marginY": (
    property: margin-top margin-bottom,
    class: my,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "marginTop": (
    property: margin-top,
    class: mt,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "marginEnd": (
    property: margin-inline-end,
    class: me,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "marginBottom": (
    property: margin-bottom,
    class: mb,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "marginStart": (
    property: margin-inline-start,
    class: ms,
    values:
      map.merge(
        $utility-spacing-vars,
        (
          auto: auto
        )
      )
  ),
  "padding": (
    property: padding,
    class: p,
    values: $utility-spacing-vars
  ),
  "paddingX": (
    property: padding-right padding-left,
    class: px,
    values: $utility-spacing-vars
  ),
  "paddingY": (
    property: padding-top padding-bottom,
    class: py,
    values: $utility-spacing-vars
  ),
  "paddingTop": (
    property: padding-top,
    class: pt,
    values: $utility-spacing-vars
  ),
  "paddingEnd": (
    property: padding-inline-end,
    class: pe,
    values: $utility-spacing-vars
  ),
  "paddingBottom": (
    property: padding-bottom,
    class: pb,
    values: $utility-spacing-vars
  ),
  "paddingStart": (
    property: padding-inline-start,
    class: ps,
    values: $utility-spacing-vars
  ),
  "color": (
    property: color,
    class: text,
    values: $utility-colors-vars
  ),
  "backgroundColor": (
    property: background-color,
    class: bg,
    values: $utility-background-colors-vars
  )
);

// Generate utilities
// @arguments: $utility , $infix, $export
// - $utility is a scope key from the utilities map.
// - $infix is defined by utility values
// - $export enables Camelcase by capitalizing the $infix
// -- This is achieved via flag using $export:"modules" (for css modules)
// This mixin generates classes with definition per a utility scope
// - Utility scopes are nested within a utility map
// - Map keys are fixed and defined within the mixin:
// --  property, class,values,
@mixin generate-utility-class($utility, $infix, $export) {
  $values: map.get($utility, values);
  @if type-of($values) == "string" or type-of(nth($values, 1)) != "list" {
    $values: list.zip($values, $values);
  }

  @each $key, $value in $values {
    $properties: map.get($utility, property);

    // Multiple properties are possible, for example with vertical or horizontal margins or paddings
    @if type-of($properties) == "string" {
      $properties: list.append((), $properties);
    }

    // Use custom class if present
    $property-class: if(map.has-key($utility, class), map.get($utility, class), list.nth($properties, 1));
    $property-class: if($property-class == null, "", $property-class);

    $infix: if($property-class == "" and string.slice($infix, 1, 1) == "-", string.slice($infix, 2), $infix);

    // Don't prefix if value key is null (eg. with shadow class)
    $property-class-modifier: if($key, if($property-class == "" and $infix == "", "", "") + $key, "");

    @if $value != null {
      @if $export == "modules" {
        $class-name: $property-class + capitalize(camelize($infix)) + capitalize(camelize($property-class-modifier));
        .#{$class-name} {
          @each $property in $properties {
            #{$property}: $value !important;
          }
        }
      } @else {
        $class-name: $property-class + $infix + -#{$property-class-modifier};
        .#{$class-name} {
          @each $property in $properties {
            #{$property}: $value !important;
          }
        }
      }
    }
  }
}

// Generate utilities
// This mixin invoke the utility mixin by setting its map and export type, and initializing the $infix.
@mixin utility-class-factory($map, $infix: "", $export: "") {
  @each $key, $utility in $map {
    // The utility can be disabled with `false`, thus check if the utility is a map first
    // Only proceed if responsive media queries are enabled or if it's the base media query
    @if type-of($utility) == "map" and ($infix == "") {
      @include generate-utility-class($utility, $infix, $export);
    }
  }
}
/*  stylelint-enable scss/at-if-no-null */
