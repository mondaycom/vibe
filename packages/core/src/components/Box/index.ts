export { default as Box, type BoxProps } from "./Box";

export * from "./Box.types";
