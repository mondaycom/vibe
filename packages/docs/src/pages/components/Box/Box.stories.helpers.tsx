import React from "react";
import { Tip } from "vibe-storybook-components";

export const TipBenefits = () => (
  <Tip title="Benefits of the Box component">Easily add Vibe styles without adding css</Tip>
);
