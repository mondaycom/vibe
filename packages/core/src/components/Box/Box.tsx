import React, { forwardRef, useRef } from "react";
import { camelCase } from "lodash-es";
import cx from "classnames";
import useMergeRef from "../../hooks/useMergeRef";
import {
  BackgroundColor as BackgroundColorEnum,
  BorderColor as BorderColorEnum,
  BoxTextColor as BoxTextColorEnum,
  Margin as MarginEnum,
  MarginBottom as MarginBottomEnum,
  MarginEnd as MarginEndEnum,
  MarginStart as MarginStartEnum,
  MarginTop as MarginTopEnum,
  MarginX as MarginXEnum,
  MarginY as MarginYEnum,
  Padding as PaddingEnum,
  PaddingBottom as PaddingBottomEnum,
  PaddingEnd as PaddingEndEnum,
  PaddingStart as PaddingStartEnum,
  PaddingTop as PaddingTopEnum,
  PaddingX as PaddingXEnum,
  PaddingY as PaddingYEnum,
  Rounded as RoundedEnum,
  Shadow as ShadowEnum,
  SizePrefixMapping
} from "./BoxConstants";
import { BackgroundColor, BorderColor, BaseBoxSize, BoxSize, BoxTextColor, RoundedSize, Shadow } from "./Box.types";
import { VibeComponent, VibeComponentProps, withStaticProps, ElementContent } from "../../types";
import styles from "./Box.module.scss";

export interface BoxProps extends VibeComponentProps {
  /**
   * The HTML element or custom component used as the root.
   */
  elementType?: keyof JSX.IntrinsicElements | string;
  /**
   * The content inside the box.
   */
  children?: ElementContent;
  /**
   * If true, the box is disabled.
   */
  disabled?: boolean;
  /**
   * If true, applies a border to the box.
   */
  border?: boolean;
  /**
   * The color of the border.
   */
  borderColor?: BorderColor;
  /**
   * The border radius of the box.
   */
  rounded?: RoundedSize;
  /**
   * The shadow style applied to the box.
   */
  shadow?: Shadow;
  /**
   * The margin applied to all sides.
   */
  margin?: BoxSize;
  /**
   * The horizontal margin.
   */
  marginX?: BoxSize;
  /**
   * The vertical margin.
   */
  marginY?: BoxSize;
  /**
   * The top margin.
   */
  marginTop?: BoxSize;
  /**
   * The end (right in LTR, left in RTL) margin.
   */
  marginEnd?: BoxSize;
  /**
   * The bottom margin.
   */
  marginBottom?: BaseBoxSize;
  /**
   * The start (left in LTR, right in RTL) margin.
   */
  marginStart?: BaseBoxSize;
  /**
   * The padding applied to all sides.
   */
  padding?: BaseBoxSize;
  /**
   * The horizontal padding.
   */
  paddingX?: BaseBoxSize;
  /**
   * The vertical padding.
   */
  paddingY?: BaseBoxSize;
  /**
   * The top padding.
   */
  paddingTop?: BaseBoxSize;
  /**
   * The end (right in LTR, left in RTL) padding.
   */
  paddingEnd?: BaseBoxSize;
  /**
   * The bottom padding.
   */
  paddingBottom?: BaseBoxSize;
  /**
   * The start (left in LTR, right in RTL) padding.
   */
  paddingStart?: BaseBoxSize;
  /**
   * The background color of the box.
   */
  backgroundColor?: BackgroundColor;
  /**
   * The text color inside the box.
   */
  textColor?: BoxTextColor;
  /**
   * If true, the box content is scrollable.
   */
  scrollable?: boolean;
  /**
   * Inline styles applied to the box.
   */
  style?: React.CSSProperties;
}

const Box: VibeComponent<BoxProps> & {
  borderColors?: typeof BorderColorEnum;
  roundeds?: typeof RoundedEnum;
  shadows?: typeof ShadowEnum;
  margins?: typeof MarginEnum;
  marginXs?: typeof MarginXEnum;
  marginYs?: typeof MarginYEnum;
  marginTops?: typeof MarginTopEnum;
  marginEnds?: typeof MarginEndEnum;
  marginBottoms?: typeof MarginBottomEnum;
  marginStarts?: typeof MarginStartEnum;
  paddings?: typeof PaddingEnum;
  paddingXs?: typeof PaddingXEnum;
  paddingYs?: typeof PaddingYEnum;
  paddingTops?: typeof PaddingTopEnum;
  paddingEnds?: typeof PaddingEndEnum;
  paddingBottoms?: typeof PaddingBottomEnum;
  paddingStarts?: typeof PaddingStartEnum;
  backgroundColors?: typeof BackgroundColorEnum;
  textColors?: typeof BoxTextColorEnum;
} = forwardRef(
  ({ className, id, elementType = "div", children, disabled, border, scrollable, style, ...props }: BoxProps, ref) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRef(ref, componentRef);

    const getClassNames = (styles: Record<string, string>, props: Record<string, any | undefined>) => {
      return Object.keys(props)
        .filter(prop => props[prop])
        .map(prop => {
          const value = props[prop];
          const prefix = SizePrefixMapping[prop as keyof typeof SizePrefixMapping];
          return prefix && typeof value === "string" ? styles[camelCase(`${prefix}-${value}`)] : styles[value];
        })
        .filter(Boolean);
    };

    return React.createElement(
      elementType,
      {
        ref: mergedRef,
        className: cx(
          styles.box,
          className,
          { [styles.opacityDisabled]: disabled, [styles.scrollable]: scrollable, [styles.border]: border },
          ...getClassNames(styles, props)
        ),
        id: id,
        style
      },
      children
    );
  }
);

export default withStaticProps(Box, {
  borderColors: BorderColorEnum,
  roundeds: RoundedEnum,
  shadows: ShadowEnum,
  margins: MarginEnum,
  marginXs: MarginXEnum,
  marginYs: MarginYEnum,
  marginTops: MarginTopEnum,
  marginEnds: MarginEndEnum,
  marginBottoms: MarginBottomEnum,
  marginStarts: MarginStartEnum,
  paddings: PaddingEnum,
  paddingXs: PaddingXEnum,
  paddingYs: PaddingYEnum,
  paddingTops: PaddingTopEnum,
  paddingEnds: PaddingEndEnum,
  paddingBottoms: PaddingBottomEnum,
  paddingStarts: PaddingStartEnum,
  backgroundColors: BackgroundColorEnum,
  textColors: BoxTextColorEnum
});
