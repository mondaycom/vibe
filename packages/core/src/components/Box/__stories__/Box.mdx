import { Meta } from "@storybook/blocks";
import { UsageGuidelines } from "vibe-storybook-components";
import {
  DIALOG_CONTENT_CONTAINER,
  DIVIDER,
  FLEX
} from "../../../storybook/components/related-components/component-description-map";
import { TipBenefits } from "./Box.stories.helpers";
import * as BoxStories from "./Box.stories";

<Meta of={BoxStories} />

# Box

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Box component is used as a wrapper component.

Its purpose is to help scaffold compositions while using Vibe's prop keys without writing new CSS. <br />It can be used as a container for
atom based compositions, it can accept all Vibe style related props and have a semantic html tag.

<Canvas of={BoxStories.Overview} />

### Import

```js
import { Box } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Use as a styled container",
    "Use to add spacing, borders, and rounded corners to areas of content",
    "Use as an inner component for spacing or styles"
  ]}
/>

<TipBenefits />

## Utility props

### BACKGROUNDS COLORS

<Canvas of={BoxStories.BackgroundColors} />

### TEXT COLORS

<Canvas of={BoxStories.TextColors} />

### BORDER

<Canvas of={BoxStories.Border} />

### BORDER COLOR

<Canvas of={BoxStories.BorderColor} />

### ROUNDED CORNERS

#### Size props

<Canvas of={BoxStories.RoundCorners} />

### SHADOW

<Canvas of={BoxStories.Shadow} />

### MARGIN

#### Size props

<Canvas of={BoxStories.Margin} />

#### Property variations per each size:

```
margin
marginX (x axis: left, right)
marginY (y axis: top, bottom)
marginTop
marginEnd
marginBottom
marginStart
```

#### Aligning Auto and none

each property variation can hold additional values for:

```
AUTO
NONE
```

### PADDING

#### Size props

<Canvas of={BoxStories.Padding} />

#### Property variations per each size:

```
padding
paddingX (x axis: left, right)
paddingY (y axis: top, bottom)
paddingTop
paddingEnd
paddingBottom
paddingStart
```

### Component Disabled

<Canvas of={BoxStories.Disabled} />

## Related components

<RelatedComponents componentsNames={[FLEX, DIVIDER, DIALOG_CONTENT_CONTAINER]} />
