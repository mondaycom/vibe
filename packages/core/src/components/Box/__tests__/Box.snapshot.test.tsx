import { vi, describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import Box from "../Box";

vi.mock("../Box.module.scss", () => {
  const styles = {
    box: "box",
    border: "border",
    borderColorUiBorderColor: "uiBorderColor",
    roundedSmall: "roundedSmall",
    shadowSmall: "shadowSmall",
    mSmall: "mSmall",
    mxSmall: "mxSmall",
    mySmall: "mySmall",
    mtSmall: "mtSmall",
    meSmall: "meSmall",
    msSmall: "msSmall",
    pSmall: "pSmall",
    pxSmall: "pxSmall",
    pySmall: "pySmall",
    ptSmall: "ptSmall",
    peSmall: "peSmall",
    pbSmall: "pbSmall",
    psSmall: "psSmall",
    bgPrimaryBackgroundColor: "bgPrimaryBackgroundColor",
    textPrimaryTextColor: "textPrimaryTextColor",
    opacityDisabled: "opacityDisabled",
    scrollable: "scrollable"
  };
  return {
    default: styles,
    ...styles
  };
});

describe("Box renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<Box></Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with children", () => {
    const tree = renderer.create(<Box>Child</Box>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with border prop", () => {
    const tree = renderer.create(<Box border />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with borderColor prop", () => {
    const tree = renderer.create(<Box borderColor="uiBorderColor" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with rounded prop", () => {
    const tree = renderer.create(<Box rounded="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with shadow prop", () => {
    const tree = renderer.create(<Box shadow="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with margin prop", () => {
    const tree = renderer.create(<Box margin="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginX prop", () => {
    const tree = renderer.create(<Box marginX="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginY prop", () => {
    const tree = renderer.create(<Box marginY="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginTop prop", () => {
    const tree = renderer.create(<Box marginTop="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginEnd prop", () => {
    const tree = renderer.create(<Box marginEnd="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with marginStart prop", () => {
    const tree = renderer.create(<Box marginStart="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with padding prop", () => {
    const tree = renderer.create(<Box padding="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingX prop", () => {
    const tree = renderer.create(<Box paddingX="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingY prop", () => {
    const tree = renderer.create(<Box paddingY="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingTop prop", () => {
    const tree = renderer.create(<Box paddingTop="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingEnd prop", () => {
    const tree = renderer.create(<Box paddingEnd="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingBottom prop", () => {
    const tree = renderer.create(<Box paddingBottom="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with paddingStart prop", () => {
    const tree = renderer.create(<Box paddingStart="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with backgroundColor prop", () => {
    const tree = renderer.create(<Box backgroundColor="primaryBackgroundColor" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with text prop", () => {
    const tree = renderer.create(<Box textColor="primaryTextColor" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with data-testid prop", () => {
    const tree = renderer.create(<Box data-testid="test-id" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
