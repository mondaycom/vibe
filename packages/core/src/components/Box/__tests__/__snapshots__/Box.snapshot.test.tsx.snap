// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box renders correctly > with backgroundColor prop 1`] = `
<div
  className="box bgPrimaryBackgroundColor"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with border prop 1`] = `
<div
  className="box border"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with borderColor prop 1`] = `
<div
  className="box uiBorderColor"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with children 1`] = `
<div
  className="box"
  data-testid="box"
>
  Child
</div>
`;

exports[`Box renders correctly > with data-testid prop 1`] = `
<div
  className="box"
  data-testid="test-id"
/>
`;

exports[`Box renders correctly > with empty props 1`] = `
<div
  className="box"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with margin prop 1`] = `
<div
  className="box mSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with marginEnd prop 1`] = `
<div
  className="box meSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with marginStart prop 1`] = `
<div
  className="box msSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with marginTop prop 1`] = `
<div
  className="box mtSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with marginX prop 1`] = `
<div
  className="box mxSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with marginY prop 1`] = `
<div
  className="box mySmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with padding prop 1`] = `
<div
  className="box pSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with paddingBottom prop 1`] = `
<div
  className="box pbSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with paddingEnd prop 1`] = `
<div
  className="box peSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with paddingStart prop 1`] = `
<div
  className="box psSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with paddingTop prop 1`] = `
<div
  className="box ptSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with paddingX prop 1`] = `
<div
  className="box pxSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with paddingY prop 1`] = `
<div
  className="box pySmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with rounded prop 1`] = `
<div
  className="box roundedSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with shadow prop 1`] = `
<div
  className="box shadowSmall"
  data-testid="box"
/>
`;

exports[`Box renders correctly > with text prop 1`] = `
<div
  className="box textPrimaryTextColor"
  data-testid="box"
/>
`;
